static/components/PoiCard.js

```diff
@@ -23,23 +24,32 @@ class PoiCard extends React.Component {
     })
     if(this.props.details._id === getHash()) {
       highlight(this.props.details._id)
-      masterCard(this.props.details._id)
+      superCard(this.props.details._id)
       firstCard(this.props.details._id)
     }
   }
 
-  calculateDistance(from, to) {
+  calculateDistance(from, to, typeDistance) {
     if(to[0] && to[1]) {
       const R = 6371e3 // metres
       const lat1 = this.state.from[0] * Math.PI / 180
       const lat2 = this.state.to[0] * Math.PI / 180
       const gapLat = ((lat2 * Math.PI / 180 )-(lat1 * Math.PI / 180))
       const gapLng = ((this.state.to[1] * Math.PI / 180)-(this.state.from[1] * Math.PI / 180))
-
       const a = Math.sin(gapLat/2) * Math.sin(gapLat/2) + Math.cos(lat1) * Math.cos(lat2) * Math.sin(gapLng/2) * Math.sin(gapLng/2)
       const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a))
 
-      return Math.trunc(R * c)
+      let distance = Math.trunc(R * c)
+      
+      if(typeDistance === 0) {
+        return distance
+      } else {
+        if(distance > 10000) {
+          distance = Math.round(distance/1000)
+          typeDistance = 'km'
+        }
+        return `${distance + ' ' + typeDistance}`
```

Comment:
WTF?!? This is worth a tweet!!!! 🤣 

You meant

    return `${distance} ${typeDistance}`